src/core/ack_event.h

```diff
@@ -0,0 +1,41 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
+
+typedef struct QUIC_ACK_EVENT {
```

Comment:
Instead of having separate files for ack_event and loss_event, I'd rather just define these at the top of congestion_control.h. Also, I'd rather not have the helper functions to create the events. Just declare them directly in the calling code.

Reply:
In my plan, there will be some optional fields in the event (something like `QUIC_SETTINGS`).
It's necessary to create a "factory function" to initialize all the field.

Reply:
But it's a bit overkill when there are only one or two callers anyways. I'd rather just have each caller initialize as necessary. If it really gets to be too much, we can refactor further to add the helper functions. Until then, the simpler (less code) the better.

Reply:
OK.